hbase-client/src/main/java/org/apache/hadoop/hbase/client/ClientAsyncPrefetchScanner.java

```diff
@@ -145,7 +148,7 @@ private void handleException() throws IOException {
     // Rethrow the exception so the application can handle it.
     while (!exceptionsQueue.isEmpty()) {
       Exception first = exceptionsQueue.peek();
-      first.printStackTrace();
+      LOG.error(first.getMessage(), first);
```

Comment:
Sorry, I'm looking into GitHub notifications only around the start of the week so I didn't manage to reply before the merge.

The recommendation is to either log an exception or rethrow it, never do both.

The reason is that you don't know how the higher levels will handle the exception. If they log it as well, you'll spam your logs with redundant entries.

In addition, logging the message and the whole exception is creating redundancy as well because the message will be contained in the stack trace.

If exceptions are logged, this should happen only in the last level. If you want to add information from this place in the call stack, throw a well-named exception contained in the same package or even class with a message providing the extra info and the original exception as the cause.

Therefore, my suggestion is to remove the logging altogether here and let the higher levels handle it as they see fit (which includes logging).